.github/workflows/root-ci-config/build_utils.py

```diff
@@ -205,6 +207,20 @@ def cmake_options_from_dict(config: Dict[str, str]) -> str:
 
     return ' '.join(output)
 
+def calc_options_hash(options: str) -> str:
+    """Calculate the hash of the options string. If "march=native" is in the
+    list of options, make the preprocessor defines resulting from it part of
+    the hash.
+    """
+    options_and_defines = options
+    if ('march_native' in options):
+        compiler_name = 'c++' if suprocess.spawn('c++') else 'clang++'
+        command = f'echo | {compiler_name} -dM -E - -march=native'
+        sp_result = subprocess.run([command], shell=True, capture_output=True, text=True)
```

Comment:
```suggestion
        sp_result = subprocess.run(shlex.split(command), capture_output=True, text=True)
```

It's best to avoid use of `shell=True` as suggested by https://docs.python.org/3/library/subprocess.html#security-considerations